composer.json

```diff
@@ -19,11 +19,11 @@
         "php": ">=5.4.0",
         "jms/serializer": "^1.7",
         "phpoption/phpoption": "^1.1.0",
-        "symfony/framework-bundle": "~2.3|~3.0"
+        "symfony/framework-bundle": "~2.3|~3.0|~4.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "^4.2|^5.0",
-        "symfony/expression-language": "~2.6|~3.0",
+        "phpunit/phpunit": "^4.8.35|^5.4.3|^6.0",
```

Comment:
We need `^6.0` to be able to install the Yaml component in version 4 (older PHPUnit versions depended on 2.x/3.x while PHPUnit 6 completely dropped the requirement of the Yaml component). However, PHPUnit 6 does not have the old `\PHPUnit_Framework_TestCase` class so that we need to change the base class of all test cases to use the `TestCase` class from the `PHPUnit\Framework` namespace. This in turn requires us to bump the dependency for PHPUnit 4 and 5 to releases that included class aliases for the namespaced `TestCase` class as a FC layer.